coretex/coretex/experiment/experiment.py

```diff
@@ -306,12 +306,21 @@ def submitMetrics(self, metricValues: Dict[str, Tuple[float, float]]) -> bool:
             True
         """
 
-        metrics = [{
-            "timestamp": time.time(),
-            "metric": key,
-            "x": value[0],
-            "y": value[1]
-        } for key, value in metricValues.items()]
+        metrics : List[Dict[str, Any]] = []
+
+        timestamp = time.time()
+
+        for key, value in metricValues.items():
+            for metric in self.metrics:
+                if metric.name == key and metric.xType == MetricType.interval.name:
```

Comment:
Can double for-loop be avoided here? You can check if metric is in dict this way `metric.name in metricValues`